File: org.kevoree.microsandbox.watchdog/scratch/PParser.cpp
/* $ANTLR 2.7.2: "P.g" -> "PParser.cpp"$ */
#include "PParser.hpp"
#include <antlr/NoViableAltException.hpp>
#include <antlr/SemanticException.hpp>
#include <antlr/ASTFactory.hpp>
#line 1 "P.g"
#line 8 "PParser.cpp"
PParser::PParser(ANTLR_USE_NAMESPACE(antlr)TokenBuffer& tokenBuf, int k)
: ANTLR_USE_NAMESPACE(antlr)LLkParser(tokenBuf,k)
{
}

PParser::PParser(ANTLR_USE_NAMESPACE(antlr)TokenBuffer& tokenBuf)
: ANTLR_USE_NAMESPACE(antlr)LLkParser(tokenBuf,1)
{
}

PParser::PParser(ANTLR_USE_NAMESPACE(antlr)TokenStream& lexer, int k)
: ANTLR_USE_NAMESPACE(antlr)LLkParser(lexer,k)
{
}

PParser::PParser(ANTLR_USE_NAMESPACE(antlr)TokenStream& lexer)
: ANTLR_USE_NAMESPACE(antlr)LLkParser(lexer,1)
{
}

PParser::PParser(const ANTLR_USE_NAMESPACE(antlr)ParserSharedInputState& state)
: ANTLR_USE_NAMESPACE(antlr)LLkParser(state,1)
{
}

void PParser::startRule() {
	
	try {      // for error handling
		{ // ( ... )+
		int _cnt21511=0;
		for (;;) {
			if ((LA(1) == INT)) {
				decl();
			}
			else {
				if ( _cnt21511>=1 ) { goto _loop21511; } else {throw ANTLR_USE_NAMESPACE(antlr)NoViableAltException(LT(1), getFilename());}
			}
			
			_cnt21511++;
		}
		_loop21511:;
		}  // ( ... )+
	}
	catch (ANTLR_USE_NAMESPACE(antlr)RecognitionException& ex) {
		reportError(ex);
		consume();
		consumeUntil(_tokenSet_0);
	}
}

void PParser::decl() {
	ANTLR_USE_NAMESPACE(antlr)RefToken  a = ANTLR_USE_NAMESPACE(antlr)nullToken;
	ANTLR_USE_NAMESPACE(antlr)RefToken  b = ANTLR_USE_NAMESPACE(antlr)nullToken;
	
	try {      // for error handling
		match(INT);
		a = LT(1);
		match(ID);
#line 35 "P.g"
		ANTLR_USE_NAMESPACE(std)cout << "decl " << a->getText() << ANTLR_USE_NAMESPACE(std)endl;
#line 69 "PParser.cpp"
		{ // ( ... )*
		for (;;) {
			if ((LA(1) == COMMA)) {
				match(COMMA);
				b = LT(1);
				match(ID);
#line 36 "P.g"
				ANTLR_USE_NAMESPACE(std)cout << "decl " << b->getText() << ANTLR_USE_NAMESPACE(std)endl;
#line 78 "PParser.cpp"
			}
			else {
				goto _loop21514;
			}
			
		}
		_loop21514:;
		} // ( ... )*
		match(SEMI);
	}
	catch (ANTLR_USE_NAMESPACE(antlr)RecognitionException& ex) {
		reportError(ex);
		consume();
		consumeUntil(_tokenSet_1);
	}
}

void PParser::initializeASTFactory( ANTLR_USE_NAMESPACE(antlr)ASTFactory& factory )
{
}
const char* PParser::tokenNames[] = {
	"<0>",
	"EOF",
	"<2>",
	"NULL_TREE_LOOKAHEAD",
	"\"int\"",
	"ID",
	"COMMA",
	"SEMI",
	"INCLUDE",
	"STRING",
	"WS_",
	0
};

const unsigned long PParser::_tokenSet_0_data_[] = { 2UL, 0UL, 0UL, 0UL };
// EOF 
const ANTLR_USE_NAMESPACE(antlr)BitSet PParser::_tokenSet_0(_tokenSet_0_data_,4);
const unsigned long PParser::_tokenSet_1_data_[] = { 18UL, 0UL, 0UL, 0UL };
// EOF "int" 
const ANTLR_USE_NAMESPACE(antlr)BitSet PParser::_tokenSet_1(_tokenSet_1_data_,4);

File: org.kevoree.microsandbox.watchdog/scratch/PParser.java
// $ANTLR 2.7.2: "P.g" -> "PParser.java"$

import antlr.TokenBuffer;
import antlr.TokenStreamException;
import antlr.TokenStreamIOException;
import antlr.ANTLRException;
import antlr.LLkParser;
import antlr.Token;
import antlr.TokenStream;
import antlr.RecognitionException;
import antlr.NoViableAltException;
import antlr.MismatchedTokenException;
import antlr.SemanticException;
import antlr.ParserSharedInputState;
import antlr.collections.impl.BitSet;

public class PParser extends antlr.LLkParser       implements PParserTokenTypes
 {

	public void traceOut(String rname) throws TokenStreamException  {
		System.out.println("exit "+rname+"; LT(1)="+LT(1));
	}
	public void traceIn(String rname) throws TokenStreamException  {
		System.out.println("enter "+rname+"; LT(1)="+LT(1));
	}
/*
	public void consume() throws IOException {
		try {
			System.out.println(LT(1));
		}
		catch (IOException ignore) {}
		super.consume();
	}
*/

protected PParser(TokenBuffer tokenBuf, int k) {
  super(tokenBuf,k);
  tokenNames = _tokenNames;
}

public PParser(TokenBuffer tokenBuf) {
  this(tokenBuf,1);
}

protected PParser(TokenStream lexer, int k) {
  super(lexer,k);
  tokenNames = _tokenNames;
}

public PParser(TokenStream lexer) {
  this(lexer,1);
}

public PParser(ParserSharedInputState state) {
  super(state,1);
  tokenNames = _tokenNames;
}

	public final void startRule() throws RecognitionException, TokenStreamException {
		
		
		try {      // for error handling
			{
			int _cnt19615=0;
			_loop19615:
			do {
				if ((LA(1)==INT)) {
					decl();
				}
				else {
					if ( _cnt19615>=1 ) { break _loop19615; } else {throw new NoViableAltException(LT(1), getFilename());}
				}
				
				_cnt19615++;
			} while (true);
			}
		}
		catch (RecognitionException ex) {
			reportError(ex);
			consume();
			consumeUntil(_tokenSet_0);
		}
	}
	
	public final void decl() throws RecognitionException, TokenStreamException {
		
		Token  a = null;
		Token  b = null;
		
		try {      // for error handling
			match(INT);
			a = LT(1);
			match(ID);
			System.out.println("decl "+a.getText());
			{
			_loop19618:
			do {
				if ((LA(1)==COMMA)) {
					match(COMMA);
					b = LT(1);
					match(ID);
					System.out.println("decl "+b.getText());
				}
				else {
					break _loop19618;
				}
				
			} while (true);
			}
			match(SEMI);
		}
		catch (RecognitionException ex) {
			reportError(ex);
			consume();
			consumeUntil(_tokenSet_1);
		}
	}
	
	
	public static final String[] _tokenNames = {
		"<0>",
		"EOF",
		"<2>",
		"NULL_TREE_LOOKAHEAD",
		"\"int\"",
		"ID",
		"COMMA",
		"SEMI",
		"INCLUDE",
		"STRING",
		"WS"
	};
	
	private static final long[] mk_tokenSet_0() {
		long[] data = { 2L, 0L};
		return data;
	}
	public static final BitSet _tokenSet_0 = new BitSet(mk_tokenSet_0());
	private static final long[] mk_tokenSet_1() {
		long[] data = { 18L, 0L};
		return data;
	}
	public static final BitSet _tokenSet_1 = new BitSet(mk_tokenSet_1());
	
	}
